CUDADataFormats/HGCal/interface/HGCConditions.h

```diff
@@ -0,0 +1,203 @@
+#ifndef CUDADataFormats_HGCal_HGCConditions_h
+#define CUDADataFormats_HGCal_HGCConditions_h
+
+#include <cstdint>
+#include <cstddef>
+#include <array>
+#include <vector>
+
+class HeterogeneousHGCSiliconDetId {
+public:
+  constexpr HeterogeneousHGCSiliconDetId(uint32_t id) : id_(id) {}
+  constexpr std::int32_t type() { return (id_ >> kHGCalTypeOffset) & kHGCalTypeMask; }
```

Comment:
I don't think there is any need to write `std::int32_t` instead of simply `int32_t`.
I would suggest removing the `std::` namespace from the integer type everywhere.